packages/arb-avm-cpp/avm/src/machinestate/machineoperation.cpp

```diff
@@ -551,30 +551,26 @@ void tlen(MachineState& m) {
 }
 
 void ec_recover(MachineState& m) {
-    m.stack.prepForMod(7);
+    m.stack.prepForMod(4);
     secp256k1_context* context = secp256k1_context_create(
         SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY);
     secp256k1_pubkey pubkey;
     secp256k1_ecdsa_recoverable_signature signature;
-    uint256_t temp;
-    for (int i = 0; i < 8; i++) {
-        auto* sig_loc = reinterpret_cast<uint64_t*>(signature.data + 8 * i);
-        temp = assumeInt(m.stack[1 - i / 4]) >> (192 - 64 * (i % 4));
-        *sig_loc = static_cast<uint64_t>(temp);
+    for (int i = 0; i < 2; i++) {
+        to_big_endian(assumeInt(m.stack[1 - i]), signature.data + (32 * i));
     }
-    signature.data[64] = static_cast<unsigned char>(assumeInt(m.stack[6]));
-    uint32_t message[32];
-    for (int i = 0; i < 32; i++) {
-        message[i] = static_cast<uint32_t>(
-            assumeInt(m.stack[5 - (i / 8)]) >> (224 - 32 * (i % 8)) &
-            std::numeric_limits<uint32_t>::max());
+    signature.data[64] = static_cast<unsigned char>(assumeInt(m.stack[3]));
+    uint32_t message[8];
+    for (int i = 0; i < 8; i++) {
+        message[i] =
+            static_cast<uint32_t>(assumeInt(m.stack[2]) >> (224 - 32 * i));
     }
```

Comment:
This should be replaceable with a single `to_big_endian` like the call above was